plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/fileio/ForwardingFileIo.java

```diff
@@ -104,35 +103,22 @@ public void deleteFiles(Iterable<String> pathsToDelete)
         partitions.forEach(this::deleteBatch);
     }
 
-    // TODO: remove below workarounds after https://github.com/apache/iceberg/pull/9953
     @Override
     public InputFile newInputFile(ManifestFile manifest)
     {
-        checkArgument(
-                manifest.keyMetadata() == null,
-                "Cannot decrypt manifest: %s (use EncryptingFileIO)",
-                manifest.path());
-        return newInputFile(manifest.path(), manifest.length());
+        return SupportsBulkOperations.super.newInputFile(manifest);
```

Comment:
I was going to remove the override entirely but I saw that in Trino there was a consensus for always explicitly overriding the interface implementation based on https://github.com/trinodb/trino/pull/21079. So I followed the same pattern here.